src/optimizer/projection_push_down_optimizer.cpp

```diff
@@ -361,8 +384,14 @@ void ProjectionPushDownOptimizer::preAppendProjection(LogicalOperator* op, idx_t
         // We don't have a way to handle
         return;
     }
-    auto projection =
-        std::make_shared<LogicalProjection>(std::move(expressions), op->getChild(childIdx));
+    auto child = op->getChild(childIdx);
+    // reset scan node table properties
+    if (child->getOperatorType() == LogicalOperatorType::SCAN_NODE_TABLE) {
```

Comment:
I would add a new optimizer that remove any unnecessary projection instead of coupling the logic inside projection push down optimizer.